src/code/z_actor.c

```diff
@@ -2046,18 +2058,18 @@ void Actor_UpdateAll(PlayState* play, ActorContext* actorCtx) {
                     refActor->world.pos.z, 0, 0, 0, 1);
     }
 
-    sp80 = &D_80116068[0];
+    canIgnoreCategoryPlayerStateFlags1 = &sCanIgnoreCategoryPlayerStateFlags1Masks[0];
 
     if (player->stateFlags2 & PLAYER_STATE2_27) {
-        unkFlag = ACTOR_FLAG_25;
+        requiredActorFlag = ACTOR_FLAG_25;
     }
 
     if ((player->stateFlags1 & PLAYER_STATE1_6) && ((player->actor.textId & 0xFF00) != 0x600)) {
         sp74 = player->targetActor;
     }
 
-    for (i = 0; i < ARRAY_COUNT(actorCtx->actorLists); i++, sp80++) {
-        unkCondition = (*sp80 & player->stateFlags1);
+    for (i = 0; i < ARRAY_COUNT(actorCtx->actorLists); i++, canIgnoreCategoryPlayerStateFlags1++) {
+        canIgnoreCategory = (*canIgnoreCategoryPlayerStateFlags1 & player->stateFlags1);
```

Comment:
So if player has any of those or'ed flags of the current actor category then update is skipped for every actor of this category?

Reply:
> So if player has any of those or'ed flags of the current actor category

yes

> then update is skipped for every actor of this category?

not exactly,
if `canIgnoreCategory` is true(ish) then actors of the category may not be updated if the condition ```(sp74 != actor) && (actor != player->naviActor) &&
                        (actor != player->heldActor) && (&player->actor != actor->parent)``` passes

the condition can be rewritten as ```!((sp74 == actor) || (actor == player->naviActor) ||
                        (actor == player->heldActor) || (&player->actor == actor->parent))```
i.e. that condition is exceptions to ignoring all the category's actors (basically, keep updating actors interacting with player it looks like)